Root/MT2_ROOT.cxx

```diff
@@ -0,0 +1,80 @@
+#include "SusyNtuple/MT2_ROOT.h"
+
+ComputeMT2::ComputeMT2(TLorentzVector visa, TLorentzVector visb, TLorentzVector MET, double ma, double mb) {
+  
+  Init(visa, visb, MET, ma, mb);
+
+}
+
+ComputeMT2::ComputeMT2(TLorentzVector visa, TLorentzVector visb, TLorentzVector MET, TString name1, TString name2){
+  if (particletype(name1) >= 0 && particletype(name2) >= 0){
+    Init(visa, visb, MET, particletype(name1), particletype(name2));
+  }
+  else{
+    std::cout << "MT2 Error: the particle names you give are not on the list (or have the wrong syntax)!" << std::endl;
+    std::cout << "Choices are: " << std::endl;
+    listparticles();
+    exit(1);
+  }
+}    
+
+void ComputeMT2::Init(TLorentzVector visa, TLorentzVector visb, TLorentzVector MET, double ma, double mb) {
+  
+  if (V1.M() < 0 || V2.M() < 0){
+    std::cout << "MT2 Error: Your input particles have negative mass" << std::endl;
+    exit(1);
+  }
+
+  V1 = visa;
+  V2 = visb;
+  ET = MET;
+  Ma = ma;
+  Mb = mb;
+  myMT2 = -1;
+  sol1 = -1;
+  sol2 = -1;
+
+}
+
+ComputeMT2::~ComputeMT2() {
+}
+
+double ComputeMT2::Compute(bool useActualVisMass, double visaMass, double visbMass)  {
```

Comment:
Original implementation doesn't take any arguments and computes MT2 using the LV.M() for the visible particles. I implemented three arguments, useActualVisMass, if set to true, give the default result (and doesn't need user to set visaMass and visbMass). If it's false, then the second and third arguments (default = 0 GeV) will be used for the masses.